src/WebAsmReader.h

```diff
@@ -0,0 +1,124 @@
+#ifndef WEBASMREADER_H
```

Comment:
And also, since this class should be able to read and write, so it does general input/output, perhaps we could call it `WasmIO`.